gateway_rejection_reason.go

```diff
@@ -0,0 +1,14 @@
+package braintree
+
+type GatewayRejectionReason string
+
+const (
+	GatewayRejectionReasonApplicationIncomplete GatewayRejectionReason = "application_incomplete"
+	GatewayRejectionReasonAVS                                          = "avs"
+	GatewayRejectionReasonAVSAndCVV                                    = "avs_and_cvv"
+	GatewayRejectionReasonCVV                                          = "cvv"
+	GatewayRejectionReasonDuplicate                                    = "duplicate"
+	GatewayRejectionReasonFraud                                        = "fraud"
+	GatewayRejectionReasonThreeDSecure                                 = "three_d_secure"
+	GatewayRejectionReasonUnrecognized                                 = "unrecognized"
```

Comment:
I've been meaning to fix up those other enums. I think we should fix them all and make them all typed. It's a breaking change but rather minor break, and until we reach v1 we've been allowing breaking changes.